x-pack/plugins/monitoring/public/components/no_data/explanations/collection_enabled/collection_enabled.js

```diff
@@ -23,11 +23,20 @@ export class ExplainCollectionEnabled extends React.Component {
   constructor(props) {
     super(props);
     this.handleClick = this.handleClick.bind(this);
+
+    this.state = {
+      waitedTooLong: false
+    };
   }
 
   handleClick() {
     const { enabler } = this.props;
     enabler.enableCollectionEnabled();
+
+    // wait 19 seconds, show link to reload
+    setTimeout(() => {
+      this.setState({ waitedTooLong: true });
```

Comment:
What happens relative to this timeout triggering if the page has navigated away automatically? Does this timer get thrown away or does it still live on with a dead reference?

Reply:
It's very hard to trigger this, but it is possible to get the warning:
![image](https://user-images.githubusercontent.com/908371/41443468-e9e72842-6ff1-11e8-9220-93f63f729cc8.png)